app/code/VConnect/Blog/Block/Widget/PostsList.php

```diff
@@ -49,10 +49,14 @@ public function getPosts(): array
 
     public function getPostUrl(Post $post): string
     {
-        return $this->urlBuilder->getUrl(
-            'vconnect_blog/post/view',
-            ['id' => $post->getData('entity_id'), '_secure' => true]
-        );
+        $baseUrl = $this->urlBuilder->getBaseUrl() . 'blog/';
+        if (!empty($post->getUrlKey()) && !is_null($post->getUrlKey())) {
+            $postUrl = $post->getUrlKey();
+        } else {
+            $postUrl = 'post-' . $post->getEntityId();
```

Comment:
try to access the post using the result of your logic. Will you be successful? www.magento.loc/post-{$id} - will you reach the post page or go to 404 ?